src/shared/components/home/signup.tsx

```diff
@@ -516,3 +565,9 @@ export class Signup extends Component<
     return `data:image/png;base64,${captcha.png}`;
   }
 }
+
+function getOAuthProvider(signup: Signup) {
+  return (signup.state.siteRes.oauth_providers || []).find(
+    provider => provider.id === Number(signup.props?.sso_provider_id || -1),
+  );
+}
```

Comment:
Use `??` instead of `||` for checking if `null` or `undefined`. Also, can `signup.state.siteRes.oauth_providers` even be undefined?